app/routes/candidates.js

```diff
@@ -26,10 +27,20 @@ router.get('/', function(req, res, next) {
     params.$district = req.query.district;
   }
 
-  db.each(selectClause + whereClause, params, function(err, row) {
-    candidates.push(row);
-  }, function() {
-    res.send(candidates);
+  db.serialize(function() {
+      var generalElection = '';
+	  // Find out if we only need to show General Election candidates
+	  db.get(selectClauseGenElect + whereClause, params, function(err, row) {
```

Comment:
so is this part of code selecting general election candidates out, then from those result do the original select? if so, would it be more clear to just make it into a single query statement?